tests/utils/git_test.py

```diff
@@ -87,176 +163,75 @@ def test_git_version_too_old(self, run_subprocess_with_logging, run_subprocess):
-    @mock.patch("esrally.utils.process.run_subprocess_with_logging")
-    def test_head_revision(self, run_subprocess_with_logging, run_subprocess):
-        run_subprocess_with_logging.return_value = 0
-        run_subprocess.return_value = ["3694a07"]
-        assert git.head_revision("/src") == "3694a07"
-        run_subprocess.assert_called_with("git -C /src rev-parse --short HEAD")
-
-    @mock.patch("esrally.utils.process.run_subprocess_with_output")
-    @mock.patch("esrally.utils.process.run_subprocess_with_logging")
-    def test_list_remote_branches(self, run_subprocess_with_logging, run_subprocess):
-        run_subprocess_with_logging.return_value = 0
-        run_subprocess.return_value = [
-            "  origin/HEAD",
-            "  origin/master",
-            "  origin/5.0.0-alpha1",
-            "  origin/5",
-        ]
-        assert git.branches("/src", remote=True) == ["master", "5.0.0-alpha1", "5"]
-        run_subprocess.assert_called_with("git -C /src for-each-ref refs/remotes/ --format='%(refname:short)'")
-
-    @mock.patch("esrally.utils.process.run_subprocess_with_output")
-    @mock.patch("esrally.utils.process.run_subprocess_with_logging")
-    def test_list_local_branches(self, run_subprocess_with_logging, run_subprocess):
-        run_subprocess_with_logging.return_value = 0
-        run_subprocess.return_value = [
-            "  HEAD",
-            "  master",
-            "  5.0.0-alpha1",
-            "  5",
-        ]
-        assert git.branches("/src", remote=False) == ["master", "5.0.0-alpha1", "5"]
-        run_subprocess.assert_called_with("git -C /src for-each-ref refs/heads/ --format='%(refname:short)'")
-
-    @mock.patch("esrally.utils.process.run_subprocess_with_output")
-    @mock.patch("esrally.utils.process.run_subprocess_with_logging")
-    def test_list_tags_with_tags_present(self, run_subprocess_with_logging, run_subprocess):
-        run_subprocess_with_logging.return_value = 0
-        run_subprocess.return_value = [
-            "  v1",
-            "  v2",
-        ]
-        assert git.tags("/src") == ["v1", "v2"]
-        run_subprocess.assert_called_with("git -C /src tag")
-
-    @mock.patch("esrally.utils.process.run_subprocess_with_output")
-    @mock.patch("esrally.utils.process.run_subprocess_with_logging")
-    def test_list_tags_no_tags_available(self, run_subprocess_with_logging, run_subprocess):
-        run_subprocess_with_logging.return_value = 0
-        run_subprocess.return_value = ""
-        assert git.tags("/src") == []
-        run_subprocess.assert_called_with("git -C /src tag")
+            git.checkout(TestGit.local_tmp_src_dir, branch=branch)
+        assert exc.value.args[0] == f"Could not checkout [{branch}]. Do you have uncommitted changes?"
+
+    def test_checkout_revision(self):
+        # Apple Git 'core.abbrev' defaults to return 7 char prefixes (09980cd)
+        # Linux defaults to return 8 char prefixes (09980cd5)
```

Comment:
I also get 7 char prefixes on my Linux laptop with git 2.38.1. I think it's enough to mention that minimum core.abbrev is 7 across all git versions

Reply:
Addressed in https://github.com/elastic/rally/pull/1602/commits/ebbb7c1e08384a2f78c5692b903caf52ff5a6e14